backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java

```diff
@@ -505,13 +506,17 @@ private ApiUser getCurrentApiUser() {
       return getCurrentApiUserNoCache();
     }
 
-    if (_apiUserContextHolder.hasBeenPopulated()) {
-      log.debug("Retrieving user from request context");
-      return _apiUserContextHolder.getCurrentApiUser();
+    try {
+      if (_apiUserContextHolder.hasBeenPopulated()) {
+        log.debug("Retrieving user from request context");
+        return _apiUserContextHolder.getCurrentApiUser();
+      }
+      ApiUser user = getCurrentApiUserNoCache();
+      _apiUserContextHolder.setCurrentApiUser(user);
+      return user;
+    } catch (ScopeNotActiveException e) {
+      return getCurrentApiUserNoCache();
```

Comment:
If I remember correctly, `getCurrentApiUserNoCache` is implemented here for tests, correct?

Reply:
In this instance, I believe it's also required for production. (You may be thinking of the changes to `DemoOktaRepository` and the `tenantDataAccessContextHolder` - those are specifically used for tests/non-Okta)